src/main/java/org/rumbledb/runtime/functions/FunctionItemCallIterator.java

```diff
@@ -212,7 +212,7 @@ private DynamicContext createNewDynamicContextWithArguments(DynamicContext conte
                 localArgumentValues.put(argName, argIterator.materialize(context));
             }
         }
-        return new DynamicContext(context, localArgumentValues, RDDArgumentValues, DFArgumentValues);
+        return new DynamicContext(null, localArgumentValues, RDDArgumentValues, DFArgumentValues);
```

Comment:
The outer dynamic context is not passed to a function call. It is taken from the closure and parameter instead.